src/main/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangAdvice.kt

```diff
@@ -0,0 +1,180 @@
+package no.nav.familie.ks.sak.sikkerhet
+
+import no.nav.familie.ks.sak.common.exception.RolleTilgangskontrollFeil
+import no.nav.familie.ks.sak.config.BehandlerRolle
+import no.nav.familie.ks.sak.config.RolleConfig
+import no.nav.familie.ks.sak.integrasjon.familieintegrasjon.IntegrasjonService
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ks.sak.kjerne.fagsak.FagsakService
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlagRepository
+import org.aspectj.lang.JoinPoint
+import org.aspectj.lang.annotation.Aspect
+import org.aspectj.lang.annotation.Before
+import org.springframework.cache.CacheManager
+import org.springframework.http.HttpMethod
+import org.springframework.stereotype.Component
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import kotlin.reflect.full.declaredMemberProperties
+
+@Aspect
+@Component
+class TilgangAdvice(
+    private val fagsakService: FagsakService,
+    private val behandlingRepository: BehandlingRepository,
+    private val personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository,
+    private val rolleConfig: RolleConfig,
+    private val integrasjonService: IntegrasjonService,
+    private val cacheManager: CacheManager,
+    private val auditLogger: AuditLogger
+) {
+
+    @Before("@annotation(sjekkTilgang) ")
+    fun sjekkTilgang(joinPoint: JoinPoint, sjekkTilgang: SjekkTilgang) {
+        when (sjekkTilgang.tilgang) {
+            Tilgang.FAGSAK -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val fagsakId = extractFieldFromJoinPoint(joinPoint, listOf("fagsakId"), sjekkTilgang.index) as Long
+                sjekkTilgangTilFagsak(fagsakId, sjekkTilgang.auditLoggerEvent)
+            }
+
+            Tilgang.BEHANDLING -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val behandlingId =
+                    extractFieldFromJoinPoint(joinPoint, listOf("behandling"), sjekkTilgang.index) as Long
+                sjekkTilgangTilBehandling(behandlingId, sjekkTilgang.auditLoggerEvent)
+            }
+
+            Tilgang.PERSON -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val personIdenter = extractFieldFromJoinPoint(
+                    joinPoint,
+                    listOf("personIdent", "søkersIdent", "ident"),
+                    sjekkTilgang.index
+                ) as String
```

Comment:
Hva skjer her? Henter man ut "personIdent" fra jsonen i restkallet? 

Hva skjer om man sender med barnasIdenter? Vil man da ikke validere om man har tilgang til barna?

Reply:
Ja, her henter man ut verdien i feltet `personIdent`,`søkersIdent` eller `ident` dersom det finnes i requestBody i restkallet. Er jo ting som dette ☝️ som gjør at annotasjons-tilnærming blir litt "iffy". Man har nå ingen måte å sende inn listen av person-identer man skal sjekke tilgangen til fra kontroller-annotasjon, så vi må hente det ut fra en request-body eller fra en path-param inne i `TilgangAdvice`. I og med at `TilgangAdvice` ikke har noe forhold til hva slags type requestBody har, er vi avhengig av at feltene vi trenger for oppslag som `fagsakId`, `behandlingsId` osv har faste predefinerte navn, som benyttes i de ulike request-objektene. 

Hadde man nå forsøkt å legge til annotasjonen på en metode som tar inn et objekt hvor identene man skal sjekke tilgang til ligger i feltet `barnaIdenter` eller `barnasIdenter` ville det blitt kastet en feil, da det er feltnavn som ikke støttes per nå. 

Slik jeg ser det er det mest sannsynlig at det er personTilgang-sjekken som blir mest påvirket av dette. Når det er sagt brukes `validerTilgangTilPersoner` direkte kun i `BehandlingController` og i `FagsakController` i ba-sak fra før. Det meste av tilgangssjekk på person skjer som en del av tilgangssjekk for fagsak og behandling.